typing_tests/merge-pine-options.ts

```diff
@@ -0,0 +1,497 @@
+	aStringOrNull = narrowDownSelectArraySingle[0].name;
+	// @ts-expect-error - test case
+	aNumber = narrowDownSelectArraySingle[0].is_of__actor.__id;
+
+	const narrowDownSelectArrayMultiple = await sdk.models.apiKey.getAll({
+		$select: ['id', 'description'],
+	});
+	aNumber = narrowDownSelectArrayMultiple[0].id;
+	aStringOrNull = narrowDownSelectArrayMultiple[0].description;
+	// @ts-expect-error - test case
+	aString = narrowDownSelectArrayMultiple[0].created_at;
+	// @ts-expect-error - test case
+	aStringOrNull = narrowDownSelectArrayMultiple[0].expiry_date;
+	// @ts-expect-error - test case
+	aStringOrNull = narrowDownSelectArrayMultiple[0].name;
+	// @ts-expect-error - test case
+	aNumber = narrowDownSelectArrayMultiple[0].is_of__actor.__id;
+
+	const narrowDownSelectArrayWithNavigation = await sdk.models.apiKey.getAll({
+		$select: ['id', 'description', 'is_of__actor'],
+	});
+	aNumber = narrowDownSelectArrayWithNavigation[0].id;
+	aStringOrNull = narrowDownSelectArrayWithNavigation[0].description;
+	aNumber = narrowDownSelectArrayWithNavigation[0].is_of__actor.__id;
+	// @ts-expect-error - test case
+	aString = narrowDownSelectArrayWithNavigation[0].created_at;
+	// @ts-expect-error - test case
+	aStringOrNull = narrowDownSelectArrayWithNavigation[0].expiry_date;
+	// @ts-expect-error - test case
+	aStringOrNull = narrowDownSelectArrayWithNavigation[0].name;
+})();
+
+// $expand properties
+await (async () => {
+	const noExpand = await sdk.models.application.getAll();
+	aNumberOrUndefined = noExpand[0].is_for__device_type?.__id;
+	aNumberOrUndefined = noExpand[0].should_be_running__release?.__id;
+	anUndefined = noExpand[0].owns__device;
+	anUndefined = noExpand[0].owns__release;
+
+	const arrayExpand = await sdk.models.application.getAll({
+		$expand: ['is_for__device_type', 'owns__device'],
+	});
+
+	aNumber = arrayExpand[0].is_for__device_type[0].id;
+	aNumberOrUndefined = arrayExpand[0].should_be_running__release?.__id;
+	aNumber = arrayExpand[0].owns__device[0].id;
+	anUndefined = arrayExpand[0].owns__release;
+
+	const objectExpand = await sdk.models.application.getAll({
+		$expand: { is_for__device_type: {}, owns__device: {} },
+	});
+
+	aNumber = objectExpand[0].is_for__device_type[0].id;
+	aNumberOrUndefined = objectExpand[0].should_be_running__release?.__id;
+	aNumber = objectExpand[0].owns__device[0].id;
+	anUndefined = objectExpand[0].owns__release;
+
+	const objectExpandWithSelect = await sdk.models.application.getAll({
+		$expand: {
+			is_for__device_type: {
+				$select: ['slug', 'name'],
+			},
+			owns__device: {},
+		},
+	});
+
+	// @ts-expect-error - test case
+	aNumber = objectExpandWithSelect[0].is_for__device_type[0].id;
+	aString = objectExpandWithSelect[0].is_for__device_type[0].name;
+	aString = objectExpandWithSelect[0].is_for__device_type[0].slug;
+
+	aNumberOrUndefined =
+		objectExpandWithSelect[0].should_be_running__release?.__id;
+	aNumber = objectExpandWithSelect[0].owns__device[0].id;
```

Comment:
NIT: Maybe better use a random other prop than `id` (eg `name`), even though we know there isn't any special handling for `id`.

Reply:
I don't think it makes sense to change the prop as you said yourself nothing is special for `id` (at least on the current approach)